bit.py

```diff
@@ -1,70 +1,246 @@
+        """
+        return len(self.sequence)
+
+    def __getitem__(self: Self, index: int | tuple) -> str:
+        """
+        Returns slice of sequence
+        Args:
+            index (int | tuple): the index to retrieve from the sequence
 
         Returns:
-        list: list with DNA or RNA sequences
-        """
-        valid_dna = set("ATGCatgc")
-        valid_rna = set("AUGCaugc")
-        lst_nucl_seq = []
-        for seq in lst_input:
-            if set(seq) <= valid_dna or set(seq) <= valid_rna:
-                lst_nucl_seq.append(seq)
-            else:
-                lst_nucl_seq.append(None)
-        return lst_nucl_seq
-
-    lst_nucl_seq = check_input(lst_input)
-
-    function_map = {
-        "transcribe": transcribe,
-        "reverse": reverse,
-        "complement": complement,
-        "reverse_complement": reverse_complement,
+            str: The nucleotide or amino acid at the specified index
+        """
+        return self.sequence.__getitem__(index)
+
+    def __repr__(self: Self) -> str:
+        """
+        Returns a “official” representation of the sequence
+        """
+        return self.sequence
+
+    def __str__(self: Self) -> str:
+        """
+        Returns the string version of the sequence
+        """
+        return str(self.sequence)
+
+    def check_correctness(self: Self) -> bool:
+        """
+        Validates the sequence against a set of characters
+
+        Returns:
+            bool: True if the sequence contains only valid characters, False otherwise
+        """
+        return set(self.sequence) <= set(self.alphabet)
+
+
+class NucleicAcidSequence(BioSequenceFunctions):
+    """
+    Class for working with nucleic acid sequences using complementary and inverse operations
+    """
+
+    alphabet = None
+    complement_dict = None
+
+    def __init__(self: Self, sequence: str) -> None:
+        """
+        Initializes a NucleicAcidSequence object
+
+        Args:
+            sequence (str): a string with nucleic acid sequence
+        """
+        self.sequence = sequence
+
+    def complement(self: Self) -> Self:
+        """
+        Returns the complement sequence
+
+        Returns:
+            NucleicAcidSequence: a new object containing the complement of the original sequence
+        """
+        complement_seq = "".join([self.complement_dict[bp] for bp in self.sequence])
+        return type(self)(complement_seq)
+
+    def reverse(self: Self) -> Self:
+        """
+        Returns the reversed sequence
+
+        Returns:
+            NucleicAcidSequence: a new sequence object with the reversed sequence
+        """
+        return type(self)(self.sequence[::-1])
+
+    def reverse_complement(self) -> Self:
+        """
+        Returns the reverse complement copy of sequence
+
+        Returns:
+            NucleicAcidSequence: a new sequence object containing the reversed complement sequence
+        """
+        self.reverse()
+        return self.complement()
```

Comment:
```suggestion
        return self.complement().reverse()
```

А то reverse вызван просто так, получается.

Минус 1 балл.